parsers/ocaml.c

```diff
@@ -1858,31 +2084,70 @@ static void clearStack ( void )
 		vStringDelete (stack[i].contextName);
 }
 
+#pragma clang diagnostic ignored "-Wattributes"
+/* clang's -licm (Loop Invariant Code Motion) optimization pass actually
+ * alters the main while loop. This has the (inadvertant and incorrect)
+ * side-effect of moving the line/filePos update statements _outside_ the loop,
+ * because it never sees them get updated inside of it ('File' is in another
+ * object file). On GCC and versions of clang bundled with LLVM 3.5 and up,
+ * the 'optnone' attribute is respected, and everything works predictably.
+ * ... I'm still going to email the clang dev team about this. */
+__attribute__ ((optnone))
```

Comment:
This should be removed, has been fixed in 585d60d5a4ea4c6f2e96c9cd65dae87fe52d25c2 (and I already tried to fix it in exuberant-ctags a long while back, see https://sourceforge.net/p/ctags/mailman/message/30813660/ for patch and details).

But anyway, before this change the compiler was actually is totally right optimizing this out.  As the `File` global was `const`, it was totally valid to assume its value never changed, leading to the behavior you experienced.  This is NOT the optimizer's fault, but our (previous) code's which used a plain hack to avoid altering the content of the `File` global.